### PR TITLE
Added block check to agent xfer channel.

### PR DESCRIPTION
**Pull request by cheftako**
Added a flag to warn us if we're going on block adding a message
to the agents xfer channel. Ran this with a kubectl cp of a 20M
file and had to bump the channel to size 150 for it not to block.

### CODE REVIEW

File: pkg/agent/clientset.go [hunk 1]
```diff
@@ -121,6 +123,7 @@ type ClientSetConfig struct {
 	SyncIntervalCap         time.Duration
 	DialOptions             []grpc.DialOption
 	ServiceAccountTokenPath string
+	WarnOnChannelLimit 		bool
```
Comment (Jefftree) [inline pkg/agent/clientset.go:126]: nit: alignment seems off, probably fixed with gofmt?

**Review by Jefftree — COMMENTED** [review]
Changes look good, but is there any reason we're doing this as a CLI flag rather than having it enabled by default on a lower log level?

**Comment by Jefftree** [comment]
Ahh please ignore my previous comment, just saw the line "The check is an "unlocked" read but is still use at your own peril."

File: cmd/agent/app/options/options.go [hunk 2]
```diff
@@ -43,6 +43,13 @@ type GrpcProxyAgentOptions struct {
 
 	// file contains service account authorization token for enabling proxy-server token based authorization
 	ServiceAccountTokenPath string
+
+	// This controls if we attempt to push onto a "full" transfer channel.
```
[the remark this answers is not in the document]
Comment (cheftako) [reply cmd/agent/app/options/options.go:47]: Done

File: cmd/agent/app/options/options.go [hunk 3]
```diff
@@ -43,6 +43,13 @@ type GrpcProxyAgentOptions struct {
...
+	//However checking that the transfer channel is full is not safe.
```
[the remark this answers is not in the document]
Comment (cheftako) [reply cmd/agent/app/options/options.go:48]: Done

File: pkg/agent/clientset.go (hunk 1 above)
Comment (cheftako) [reply pkg/agent/clientset.go:126]: Done

**Review by rata — APPROVED** [review]
LGTM :)

**Pull request merged**